Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/FamilyTestRepository.kt

```diff
@@ -1,79 +1,89 @@
 package de.rki.coronawarnapp.familytest.core.repository
 
 import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestQRCode
+import de.rki.coronawarnapp.coronatest.server.CoronaTestResult
 import de.rki.coronawarnapp.coronatest.type.TestIdentifier
+import de.rki.coronawarnapp.familytest.core.model.CoronaTest
 import de.rki.coronawarnapp.familytest.core.model.FamilyCoronaTest
 import de.rki.coronawarnapp.familytest.core.model.markBadgeAsViewed
 import de.rki.coronawarnapp.familytest.core.model.markDccCreated
 import de.rki.coronawarnapp.familytest.core.model.markViewed
-import de.rki.coronawarnapp.familytest.core.model.recycle
+import de.rki.coronawarnapp.familytest.core.model.moveToRecycleBin
 import de.rki.coronawarnapp.familytest.core.model.restore
+import de.rki.coronawarnapp.familytest.core.model.updateLabId
 import de.rki.coronawarnapp.familytest.core.model.updateResultNotification
+import de.rki.coronawarnapp.familytest.core.model.updateSampleCollectedAt
+import de.rki.coronawarnapp.familytest.core.model.updateTestResult
 import de.rki.coronawarnapp.familytest.core.storage.FamilyTestStorage
-import de.rki.coronawarnapp.tag
 import de.rki.coronawarnapp.util.TimeStamper
-import de.rki.coronawarnapp.util.coroutine.AppScope
-import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
-import de.rki.coronawarnapp.util.flow.shareLatest
-import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.plus
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class FamilyTestRepository @Inject constructor(
-    dispatcherProvider: DispatcherProvider,
-    @AppScope private val appScope: CoroutineScope,
     private val processor: BaseCoronaTestProcessor,
     private val storage: FamilyTestStorage,
     private val timeStamper: TimeStamper,
 ) {
 
-    private val familyTestMap = storage.familyTestMap
-
     val familyTests: Flow<Set<FamilyCoronaTest>> = storage.familyTestMap.map {
-        it.values.filter { test -> test.isNotRecycled }.toSet()
-    }.shareLatest(
-        tag = TAG,
-        scope = appScope + dispatcherProvider.IO
-    )
-
-    val familyTestRecycleBin: Flow<Set<FamilyCoronaTest>> = storage.familyTestMap.map {
-        it.values.filter { test -> test.isRecycled }.toSet()
-    }.shareLatest(
-        tag = TAG,
-        scope = appScope + dispatcherProvider.IO
-    )
```

Comment:
`shareLatest` shares one flow with all subscribers, any reason not to?

Reply:
yes, it is not necessary and brakes unit test.

Reply:
runBlockingTest2(ignoreActive=true) let's anything pass, that's why I also removed it from RiskResultStorage, those tests were useless due to that... I do not think that shareLatest is an improvement overall, has too many complications